app/models/runtime/constraints/health_check_policy.rb

```diff
@@ -1,14 +1,29 @@
+require 'models/helpers/health_check_types'
+require 'utils/uri_utils'
+
 class HealthCheckPolicy
-  def initialize(process, health_check_timeout, health_check_invocation_timeout)
+  def initialize(process, health_check_timeout, health_check_invocation_timeout, health_check_type, health_check_http_endpoint)
```

Comment:
We refactored the health check policy, to contain all of the checks for health checks, and removed some checks that were in the process model. We think this consolidates these checks more. 

Also we refactored so that we could make a ReadinessHealthCheckPolicy that inherits from this class.